lib/porky_lib/symmetric.rb

```diff
@@ -191,6 +191,40 @@ def secure_delete_plaintext_key(length)
     "\0" * length
   end
 
+  def encrypt_with_key(plaintext, plaintext_key)
+    # Initialize the box
+    secret_box = RbNaCl::SecretBox.new(plaintext_key)
+
+    # First, make a nonce: A single-use value never repeated under the same key
+    # The nonce isn't secret, and can be sent with the ciphertext.
+    # The cipher instance has a nonce_bytes method for determining how many bytes should be in a nonce
+    nonce = RbNaCl::Random.random_bytes(secret_box.nonce_bytes)
+
+    # Encrypt a message with SecretBox
+    ciphertext = secret_box.encrypt(nonce, plaintext)
+
+    result = OpenStruct.new
+
+    result.ciphertext = ciphertext
+    result.nonce = nonce
+
+    result
+  end
+
+  def decrypt_with_key(ciphertext, plaintext_key, nonce)
+    # Initialize the box
+    secret_box = RbNaCl::SecretBox.new(plaintext_key)
+
+    # Decrypt the message
+    plaintext = secret_box.decrypt(nonce, ciphertext)
+
+    result = OpenStruct.new
+
+    result.plaintext = plaintext
```

Comment:
I'm setting this to be a struct with one value for extensibility, should one day we need to return more than just the plaintext I want to be able to add it to the struct being returned without having to change every single client that calls this function.